bundles/org.openhab.binding.openthermgateway/src/main/java/org/openhab/binding/openthermgateway/internal/Message.java

```diff
@@ -0,0 +1,244 @@
+     * R: Request sent to the boiler
+     * A: Response returned to the thermostat
+     * E: Parity or stop bit error
+     */
+
+    private String code;
+    private MessageType messageType;
+    private int id;
+    private String data;
+
+    public String getCode() {
+        return this.code;
+    }
+
+    public MessageType getMessageType() {
+        return messageType;
+    }
+
+    public int getID() {
+        return id;
+    }
+
+    public @Nullable String getData(ByteType byteType) {
+        if (this.data.length() == 4) {
+            switch (byteType) {
+                case HighByte:
+                    return this.data.substring(0, 2);
+                case LowByte:
+                    return this.data.substring(2, 4);
+                case Both:
+                    return this.data;
+            }
+        }
+
+        return null;
+    }
+
+    public boolean getBit(ByteType byteType, int pos) {
+        @Nullable
+        String data = getData(byteType);
+
+        if (data != null) {
+            // First parse the hex value to an integer
+            int parsed = Integer.parseInt(data, 16);
+
+            // Then right shift it pos positions so that the required bit is at the front
+            // and then apply a bitmask of 00000001 (1)
+            return ((parsed >> pos) & 1) == 1;
+        }
+
+        return false;
+    }
+
+    public int getUInt(ByteType byteType) {
+        @Nullable
+        String data = getData(byteType);
+
+        if (data != null) {
+            return Integer.parseInt(data, 16);
+        }
+
+        return 0;
+    }
+
+    public int getInt(ByteType byteType) {
+        @Nullable
+        String data = getData(byteType);
+
+        if (data != null) {
+            return parseSignedInteger(data);
+        }
+
+        return 0;
+    }
+
+    public float getFloat() {
+        // f8.8, two's complement
+        @Nullable
+        String data = getData(ByteType.Both);
+
+        if (data != null) {
+            long value = Long.parseLong(data, 16);
+
+            // left padded with zeros
+            String binary = String.format("%16s", Long.toBinaryString(value)).replace(' ', '0');
+
+            if (binary.charAt(0) == '1') {
+                // negative value
+
+                String inverted = invertBinary(binary);
+
+                value = Long.parseLong(inverted, 2);
+                value = value + 1;
+                value = value * -1;
+            }
+
+            // divide by 2^8 = 256
+            return (float) value / 256;
+        }
+
+        return 0;
+    }
+
+    public boolean overrides(@Nullable Message other) {
+        // If the message is a Request sent to the boiler or an Answer returned to the
+        // thermostat, and it's ID is equal to the previous message, then this is an
+        // override sent by the OpenTherm Gateway
+        if (other != null && this.getID() == other.getID() && (this.getCode() == "R" || this.getCode() == "A")) {
```

Comment:
Don't compare strings with `==`:
```suggestion
        return other != null && this.getID() == other.getID() && ("R".equals(this.getCode()) || "A".equals(this.getCode()));
```
And just return the condition.

Reply:
Adjusted